x-pack/plugins/apm/server/lib/services/get_services/get_services_items_stats.ts

```diff
@@ -0,0 +1,309 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { arrayUnionToCallable } from '../../../../common/utils/array_union_to_callable';
+import {
+  PROCESSOR_EVENT,
+  TRANSACTION_DURATION,
+  AGENT_NAME,
+  SERVICE_ENVIRONMENT,
+} from '../../../../common/elasticsearch_fieldnames';
+import { mergeProjection } from '../../../../common/projections/util/merge_projection';
+import { ProcessorEvent } from '../../../../common/processor_event';
+import {
+  ServicesItemsSetup,
+  ServicesItemsProjection,
+} from './get_services_items';
+
+const MAX_NUMBER_OF_SERVICES = 500;
+
+const getDeltaAsMinutes = (setup: ServicesItemsSetup) =>
+  (setup.end - setup.start) / 1000 / 60;
+
+interface AggregationParams {
+  setup: ServicesItemsSetup;
+  projection: ServicesItemsProjection;
+}
+
+export const getTransactionDurationAverages = async ({
+  setup,
+  projection,
+}: AggregationParams) => {
+  const { client, indices } = setup;
+
+  const response = await client.search(
+    mergeProjection(projection, {
+      size: 0,
+      index: indices['apm_oss.transactionIndices'],
+      body: {
+        query: {
+          bool: {
+            filter: projection.body.query.bool.filter.concat({
+              term: {
+                [PROCESSOR_EVENT]: ProcessorEvent.transaction,
+              },
+            }),
+          },
+        },
+        aggs: {
+          services: {
+            terms: {
+              ...projection.body.aggs.services.terms,
```

Comment:
Not related to this PR specifically but I want to invite you into my head and how I try to reason about the projections :)

I wonder what these terms are so I try to search for them. If I search for `projection.body.aggs.services.terms`, `body.aggs.services.terms`, `aggs.services.terms` or `services.terms` I don't get any results.

<img src="https://user-images.githubusercontent.com/209966/86366244-d7285c80-bc7a-11ea-811b-58ce93096b2f.png" width=400/>


Then I search for `projection` and get a lot results:
<img src="https://user-images.githubusercontent.com/209966/86366187-c37cf600-bc7a-11ea-8cde-48880d2bef3f.png" width=400/>

Then I go into manual mode and search for `getTransactionDurationAverages` which leads me to `getServicesItems` which is calling `getServicesProjection` where I eventually find:

<img src="https://user-images.githubusercontent.com/209966/86366644-51f17780-bc7b-11ea-8eb3-a5846e240503.png" width="600">

I can now see that the `terms` I'm looking for are on line 46. Great! But I notice it's a ternary so I have to back up a bit and figure out how `getServicesProjection` is called. I now see that `getServicesProjection` is called with `noEvents=true` so the actual line I'm looking for is 42 : `[]`.

It's definitely possible to find these things but it requires patience and focus and not super easy on Github (would have been easier in an editor).

Reply:
![Screen Recording 2020-07-02 at 15 57 40](https://user-images.githubusercontent.com/352732/86368060-0b9d1800-bc7d-11ea-9d7e-8665870ea220.gif)
 😝

Reply:
Haha, nice. Wish I could do that on Github. And I now realise that the value I found was wrong (looking in the query instead of the agg 😳)

Reply:
VSCode's github plugin might help btw, have you tried it?

Reply:
I have -  a while ago. I should try it again